sched/test_ratio_static.py

```diff
@@ -0,0 +1,241 @@
+"""
+Test for ratio static scheduler. All servers has the same (default) weight.
+
+Load must be distributed uniformly across all servers in the group.
+All servers must receive almost the same number of requests, less than 0.5%
+error is allowed. Enforce allowed error to 10 requests for short term tests.
+
+Number of server connections doesn't affect load distribution, so test both:
+- the same number of connections between all servers.
+- random number of connections for each server.
+
+Backend is configured to never close connections (keepalive_requests),
+since it unpredictably affects load distribution.
+"""
+
+from framework import tester
+from helpers.control import servers_get_stats
+from helpers import tf_cfg
+
+__author__ = 'Tempesta Technologies, Inc.'
+__copyright__ = 'Copyright (C) 2018 Tempesta Technologies, Inc.'
+__license__ = 'GPL2'
+
+NGINX_CONFIG = """
+pid ${pid};
+worker_processes  auto;
+
+events {
+    worker_connections   1024;
+    use epoll;
+}
+
+http {
+    keepalive_timeout ${server_keepalive_timeout};
+    keepalive_requests 1000000000;
+    sendfile         on;
+    tcp_nopush       on;
+    tcp_nodelay      on;
+
+    open_file_cache max=1000;
+    open_file_cache_valid 30s;
+    open_file_cache_min_uses 2;
+    open_file_cache_errors off;
+
+    # [ debug | info | notice | warn | error | crit | alert | emerg ]
+    # Fully disable log errors.
+    error_log /dev/null emerg;
+
+    # Disable access log altogether.
+    access_log off;
+
+    server {
+        listen        ${server_ip}:${port};
+
+        location / {
+            root ${server_resources};
+        }
+        location /nginx_status {
+            stub_status on;
+        }
+    }
+}
+"""
+
+TEMPESTA_CONFIG = """
+cache 0;
+server ${server_ip}:8000;
+server ${server_ip}:8001;
+server ${server_ip}:8002;
+server ${server_ip}:8003;
+server ${server_ip}:8004;
+server ${server_ip}:8005;
+server ${server_ip}:8006;
+server ${server_ip}:8007;
+server ${server_ip}:8008;
+server ${server_ip}:8009;
+
+"""
+
+# Random number of connections for each server.
+TEMPESTA_CONFIG_VAR_CONNS = """
+cache 0;
+server ${server_ip}:8000;
+server ${server_ip}:8001 conns_n=5;
+server ${server_ip}:8002 conns_n=30;
+server ${server_ip}:8003 conns_n=6;
+server ${server_ip}:8004 conns_n=12;
+server ${server_ip}:8005 conns_n=33;
+server ${server_ip}:8006 conns_n=4;
+server ${server_ip}:8007 conns_n=2;
+server ${server_ip}:8008 conns_n=10;
+server ${server_ip}:8009 conns_n=12;
+
+"""
+
+def sched_ratio_static_def_weight():
+    return 50
```

Comment:
Looks like this function is redundant, since it is not used in this file.

Reply:
Fixed.